ports.go

```diff
@@ -43,6 +43,8 @@ type BondData struct {
 // Port is a hardware port associated with a reserved or instantiated hardware
 // device.
 type Port struct {
+	*Href `json:",inline"`
```

Comment:
I noticed that the Port field did not include the Href field, so I included it by inlining the Href type.
This is not the same as `Href string `json:"omitempty,href"`.   
While we do get Port.Href, We get it via Port.Href.Href. Type embedding allows for an Href method to be applied on Port.  If we make:
```
func (h *Href) GetID() { return filepath.Base(h.Href) }
```
we can call `vlanAssignment.Port.GetID()`. 

This is in part inspired by and in preparation for https://github.com/packethost/packngo/pull/280/files#diff-d5c3dbfdac6c7539c18385ddd9c9dceff98dfda42c0bc25acaf87135caf6e066R31.  With the hydration PR, this would also make it possible to `vlanAssignment.Port.Hydrate()` to fill in the fields if the Port was not included.  Since nearly all resources include Href, this gives us a generic interface that we can use for common operations (and will allow us to further DRY). 

If the port was not present (`"port": {"href": ""}`) in the response, `vlanAssignment.Port` will be `nil`.
If href was omitted: `"port": {"id": "", ... (no href)}`, `Href` will be nil.

Most resources can be represented as an Href.  Rather than say "VLANAssignment.Port is an Href", I'm saying "VLANAssignment.Port is a Port which has an Href".  This makes the Port resource available when `includes=port`, but we will at least get the Href (string) when not included.

Reply:
I'll see if I can clear this up with some tests as you've asked for.

Reply:
OK, sounds good. I am up for the Hydration, I just didn't make the connection.